src/sync.cpp

```diff
@@ -5,6 +5,7 @@
 #include <sync.h>
 
 #include <logging.h>
+#include <util/memory.h>
```

Comment:
In this case, I think it was already unnecessary to include, and good time to remove.

1. ```#include <memory>``` has been added to ```sync.cpp``` because it already using ```std::unique_ptr``` ( #12859).
2. ```std::unique_ptr``` has been removed from ```sync.cpp``` (#13300)